<?xml version="1.0"?>
<launch>

    <!-- <args for urdf/> -->
    <arg name="camera_model"         default="BW1098OAK" /> <!-- 'zed' or 'zedm' -->
    <arg name="camera_name"          default="oak" />

    <arg name="base_frame"           default="base_link" />

    <arg name="cam_pos_x"             default="0.0" />    <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_y"             default="0.0" />    <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_pos_z"             default="0.0" />    <!-- Position respect to base frame (i.e. "base_link) -->
    <arg name="cam_roll"              default="1.5708" /> <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_pitch"             default="0.0" />    <!-- Orientation respect to base frame (i.e. "base_link) -->
    <arg name="cam_yaw"               default="1.5708" /> <!-- Orientation respect to base frame (i.e. "base_link) -->

    <arg name="camera_param_uri"      default="package://oak_interface/params/camera" />

 
    <include file="$(find oak_interface)/launch/urdf.launch">
        <arg name="base_frame"      value="$(arg  base_frame)"  />
        <arg name="camera_model"    value="$(arg  camera_model)"/>
        <arg name="camera_name"     value="$(arg  camera_name)" />
        <arg name="cam_pos_x"       value="$(arg  cam_pos_x)"   />
        <arg name="cam_pos_y"       value="$(arg  cam_pos_y)"   />
        <arg name="cam_pos_z"       value="$(arg  cam_pos_z)"   />
        <arg name="cam_roll"        value="$(arg  cam_roll)"    />
        <arg name="cam_pitch"       value="$(arg  cam_pitch)"   />
        <arg name="cam_yaw"         value="$(arg  cam_yaw)"     />
    </include>

    <!-- Cameras -->
    <param name="publish_mono"      type="bool" value="false" />
    <param name="publish_depth"     type="bool" value="true" />
    <param name="publish_rectified" type="bool" value="false" />
    <param name="publish_color"     type="bool" value="false" />

    <!-- Cameras parameters -->
    <param name="mono_camera_resolution" type="int"    value="720"  /> <!-- 720, 800, 400 -->
    <param name="mono_camera_fps"        type="double" value="60.0" /> <!-- Max 120-->

    <param name="color_camera_resolution" type="int"    value="1080"  /> <!-- 1080, 3840 (4K), 2826 (12MP) -->
    <param name="color_camera_fps"        type="double" value="60.0" />  <!-- Max 60-->
    <param name="use_BGR"                 type="bool"   value="true" /> <!-- Default: RGB -->
    <param name="color_interleaved"       type="bool"   value="true" />
    <param name="image_orientation"       type="int"    value="1" /> <!-- Possible values: 1-AUTO, 2-NORMAL, 3-HORIZONTAL_MIRROR, 4-VERTICAL_FLIP, 5-ROTATE_180_DEG -->


    <!-- IMU -->
    <param name="publish_imu"       type="bool" value="true" />
       <group ns="imu_covariance">
        <rosparam file="$(find oak_interface)/params/imu/imu.yaml" />
    </group>


    <!-- Neural Network -->
    <param name="NN_input_width"     type="int" value="300" />
    <param name="NN_input_high"      type="int" value="300" />
    <param name="publish_color_detections"  type="bool" value="false" />
    <param name="publish_stereo_detections" type="bool" value="false" /> 
    <param name="nnBlobPath"         type="string" value="$(find oak_interface)/resources/mobilenet-ssd_openvino_2021.2_6shave.blob" />

    <!-- Node -->
    <node name="oak" pkg="oak_interface" type="oak" output="screen" required="true">
        <param name="camera_name" value="$(arg camera_name)"/>
        <param name="camera_param_uri" value="$(arg camera_param_uri)"/>

        <remap from="left/image"  to="/oak/left/image"/>
        <remap from="left/info"   to="/oak/left/info"/>
        <remap from="right/image" to="/oak/right/image"/>
        <remap from="right/info"  to="/oak/right/info"/>

        <remap from="left_rectified/image"  to="/oak/left_rectified/image"/>
        <remap from="left_rectified/info"   to="/oak/left_rectified/info"/>
        <remap from="right_rectified/image" to="/oak/right_rectified/image"/>
        <remap from="right_rectified/info"  to="/oak/right_rectified/info"/>

        <remap from="depth/image"  to="/oak/depth/image"/>
        <remap from="depth/info"   to="/oak/depth/camera_info"/>

        <remap from="color/image"  to="/oak/color/image"/>
        <remap from="color/info"   to="/oak/color/info"/>

        <remap from="/detections_images"  to="/oak/detections/image"/>
        <remap from="/detections_depth"   to="/oak/detections/depth"/>

        <remap from="/detections_image_right"    to="/oak/detections/image/right"/>
        <remap from="/detections_image_left"    to="/oak/detections/image/left"/>

        <remap from="Imu"        to="/oak/imu"/>
        
    </node>

    <!-- Show all images -->
    
    
    

    <!-- Show color detections images -->.
    <!-- <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_detections" args="/oak/detections/image"/> -->

    <!-- Show depth detections images -->
    <!--
    <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_left_dets" args="/oak/detections/image/left"/>
    <node pkg="rqt_image_view" type="rqt_image_view" name="image_view_right_dets" args="/oak/detections/image/right"/>
     -->


</launch>
